lib/mayaUsd/ufe/UsdAttribute.cpp

```diff
@@ -380,149 +331,25 @@ namespace ufe {
 // UsdAttribute:
 //------------------------------------------------------------------------------
 
-#ifdef UFE_V4_FEATURES_AVAILABLE
-UsdAttribute::UsdAttribute(const PXR_NS::UsdPrim& prim, const Ufe::AttributeDef::ConstPtr& attrDef)
-    : fPrim(prim)
-    , fAttrDef(attrDef)
-    , fUsdAttr(prim.GetAttribute(PXR_NS::UsdShadeUtils::GetFullName(
-          TfToken(attrDef->name()),
-          attrDef->ioType() == Ufe::AttributeDef::INPUT_ATTR ? PXR_NS::UsdShadeAttributeType::Input
-                                                             : PXR_NS::UsdShadeAttributeType::Output
-
-          )))
-{
-    PXR_NAMESPACE_USING_DIRECTIVE
-    TF_VERIFY(
-        fPrim.IsValid(),
-        "Invalid prim '%s' passed to UsdAttribute object",
-        fPrim.GetName().GetText());
-}
-#endif
-
-UsdAttribute::UsdAttribute(const PXR_NS::UsdAttribute& usdAttr)
-    : fPrim(usdAttr.GetPrim())
-    , fUsdAttr(usdAttr)
+UsdAttribute::UsdAttribute(UsdAttributeHolder* attrHolder)
```

Comment:
Because the UsdAttribute class owns that pointer. That is why the holder is declared as `std::unique_ptr<UsdAttributeHolder>` in the UsdAttribute class.

Reply:
Done. If I am not mistaken, correctly indicating intent from the moment of creation requires passing the unique_ptr by rvalue reference, with a few std::move to keep it rvalued.

Reply:
Rather: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-consume